public/javascripts/songForm.js

```diff
@@ -0,0 +1,264 @@
+                    ]),
+                    m('.form-group', [
+                        m('label', 'Author'),
+                        m('input.form-control', {
+                            value: author(),
+                            onchange: m.withAttr('value', author),
+                            config: function(elem, isInit) {
+                                if (!isInit) {
+                                    if (readonly) {
+                                        $(elem).prop('readonly', true)
+                                    }
+                                }
+                            }
+                        })
+                    ]),
+                    m('.form-group', [
+                        m('label', 'Year Published'),
+                        m('input.form-control', {
+                            value: year(),
+                            onchange: m.withAttr('value', year),
+                            config: function(elem, isInit) {
+                                if (!isInit) {
+                                    if (readonly) {
+                                        $(elem).prop('readonly', true)
+                                    }
+                                }
+                            }
+                        })
+                    ]),
+                    m('.form-group', [
+                        m('label', 'Language'),
+                        m('select.form-control.capitalize', {
+                            onchange: m.withAttr('value', lang),
+                            config: function(elem, isInit) {
+                                if (!isInit) {
+                                    $(elem).val(lang())
+                                    if (readonly) {
+                                        $(elem).prop('disabled', true)
+                                    }
+                                }
+                            }
+                        }, [
+                            LANGUAGE_OPTIONS.map((lang) => {
+                                return m('option', lang)
+                            })
+                        ])
+                    ]),
+                    m('.form-group', [
+                        m('label', 'Copyright'),
+                        m('select.form-control.capitalize', {
+                            onchange: m.withAttr('value', copyright),
+                            config: function(elem, isInit) {
+                                if (!isInit) {
+                                    $(elem).val(copyright())
+                                    if (readonly) {
+                                        console.log('hehehe')
+                                        $(elem).prop('disabled', true)
+                                    }
+                                }
+                            }
+                        }, [
+                            m('option', 'CC0'),
+                            m('option', 'public')
```

Comment:
CC0
Public domain

Maybe include a ? icon next to it that has a brief explanation when people hover or tap so they know the difference.